pkg/apis/admissionregistration/types.go

```diff
@@ -115,6 +115,234 @@ const (
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
+// ValidatingAdmissionPolicy describes the definition of an admission validation policy that accepts or rejects an object without changing it.
+type ValidatingAdmissionPolicy struct {
+	metav1.TypeMeta
+	// Standard object metadata; More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata.
+	// +optional
+	metav1.ObjectMeta
+	// Specification of the desired behavior of the ValidatingAdmissionPolicy.
+	Spec ValidatingAdmissionPolicySpec
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ValidatingAdmissionPolicyList is a list of ValidatingAdmissionPolicy.
+type ValidatingAdmissionPolicyList struct {
+	metav1.TypeMeta
+	// Standard list metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+	// +optional
+	metav1.ListMeta
+	// List of ValidatingAdmissionPolicy.
+	Items []ValidatingAdmissionPolicy
+}
+
+// ValidatingAdmissionPolicySpec is the specification of the desired behavior of the AdmissionPolicy.
+type ValidatingAdmissionPolicySpec struct {
+	// ParamSource specifies the kind of resources used to parameterize this policy which has to be a CRD.
```

Comment:
Comment updated. The followup work will be tracked through following implementation work

Reply:
Question: would this authz check be a blocker for the first alpha delivery? Would some kind of constraints to prevent user from matching to resources like secret good enough? (I am not quite familiar with how to check the user access at this point. Any reference would be appreciated!)

Reply:
breadcrumbs for prior art on authz checks:
* done in admission with restmapper resolution - plugin/pkg/admission/gc/gc_admission.go
* done in API type storage directly - pkg/registry/rbac/clusterrole/policybased/storage.go in Create/Update as authz check wrapper before delegating to normal storage Create/Update

I think we need to at least understand the following before alpha:
* will we do authz checks on Policy (would have to check for global read access on the specified type because we don't have name/namespace coordinates yet)?
    * what happens if the restmapping is unresolveable (e.g. points to CRD before or concurrent with CRD creation so the kind → resource mapping isn't in discovery yet)
* will we do authz checks on binding (probably so)
    * what happens if the policy is unresolveable?
    * what happens if the restmapping is unresolveable (e.g. points to CRD before or concurrent with CRD creation so the kind → resource mapping isn't in discovery yet)?
    * what happens if the policy paramSource apiVersion/kind changes after the policybinding is created?
* would this be done in the storage layer like RBAC so it's non-optional?
* what escape hatches will we have for cluster-admins or explicitly authorized use (like the RBAC superuser or `escalate`/`bind` permissions?